### DISCUSSION TITLE
Add ability to change background dim color in grayscale

### DISCUSSION THREAD

**Discussion by santerinogelainen** [ppy/osu#12983] [Ideas] [answered]
A slider in settings that changes the lightness of the background dim color. I doubt adding a full color picker is necessary 

rrtyui used to play with a white/grayish bg.

**Comment by peppy** [comment]
This may require a custom shader to avoid +1x overdraw.

**Comment by Blifaloo** [comment]
It should be a custom background that can be enabled and disabled.

**Comment by Uncomfy** [comment]
I have tried implementing it with a custom shader for osu.Game.Graphics.Backgrounds.BeatmapBackground class ([here's the code](https://github.com/Uncomfy/osu/tree/background_dim_colour), it relies on updated [osu-resources](https://github.com/Uncomfy/osu-resources/tree/background_dim_colour)), however, it has downsides:
1. Doesn't work for maps with storyboards/videos
2. Forces BufferedContainer to redraw on dimming change

Does it make sense to update osu.Framework.Graphics.Rendering.RenderingExtensions.DrawQuad to support ColourOffset? That would make implementing background dim color trivial in osu.Game.Graphics.Containers.UserDimContainer class.
But I wonder if this is too big of an ask just for this feature

**Reply by peppy** [reply]
I'd probably sooner see it happening with a custom local shader.

**Reply by Uncomfy** [reply]
Would it be acceptable if it only works for regular backgrounds, but not for storyboards and videos? It seems like blur doesn't work on those either (though I guess it's because of the performance concerns).

Another question is what would be the most preferred way of it being implemented:
1. Only the background sprite's shader is changed, but BufferedContainer needs to be redrawn after any changes in dimming
2. Only the BufferedContainer's shader is changed, but then it would need to be automatically added to handle dimming
3. Shaders for both the background sprite and BufferedContained are changed, but then it would mean that two different things are responsible for handling dimming depending on context
4. Some other way I'm missing?

**Comment by Joehuu** [comment, marked as answer]
https://github.com/ppy/osu/pull/30391
